src/actions/GameActions.tsx

```diff
@@ -1,20 +1,39 @@
+import { EffectName, StatusEffect } from 'src/GamePlay/Effect';
+import { StoreState } from 'src/types/StoreState';
+import { TypedAction } from "redoodle";
 import { ThunkAction } from 'redux-thunk';
-import { ofType, unionize, UnionOf } from 'unionize';
-import { EffectName } from 'src/GamePlay/Effect';
-import { StoreState, Enemy } from 'src/types/StoreState';
-
-export const Actions = unionize({
-    ADJUST_HP: ofType<{ targetEntityId: string, hp: number }>(),
-    ADJUST_ENERGY: ofType<{ energy: number }>(),
-    SET_ENERGY: ofType<{ energy: number }>(),
-    APPLY_EFFECT: ofType<{ effectName: EffectName, targetId: string, magnitude: number }>(),
-    CLEAR_ENEMIES: ofType<{}>(),
-    ADD_ENEMY: ofType<Partial<Enemy>>(),
-    REMOVE_ENEMY: ofType<string>(),
-}, {
-    tag: 'type',
-    value: 'payload',
-});
-
-export type ActionsType = UnionOf<typeof Actions>;
-export type ThunkType = ThunkAction<void, StoreState, void, ActionsType>
\ No newline at end of file
+
+export const AdjustHp = TypedAction.define("gameplay::adjustHp")<{ 
+    targetEntityId: string;
+    hp: number;
+}>();
+
+export const AdjustEnergy = TypedAction.define("gameplay::adjustEnergy")<{
+     energy: number; 
+}>();
+
+export const SetEnergy = TypedAction.define("gameplay::setEnergy")<{
+    energy: number;
+}>();
+
+export const ApplyEffect  = TypedAction.define("gameplay::applyEffect")<{
+    effectName: EffectName;
+    targetId: string;
+    magnitude: number;
+}>();
+
+export const ClearEnemies  = TypedAction.define("gameplay::clearEnemies")<{
+}>();
+
+export const AddEnemy  = TypedAction.define("gameplay::addEnemy")<{
+    hp: number;
+    maxHp: number;
+    effectList: StatusEffect[];
+}>();
+
+export const RemoveEnemy  = TypedAction.define("gameplay::removeEnemy")<{
+    id: string;
+}>();
+
+// TODO: Last any was supposed to be union of types...we don't have access to?
+export type ThunkType = ThunkAction<void, StoreState, void, any>
```

Comment:
Hmmmm, I'm not sure what exactly that last generic is supposed to be. It is "the type of the Action that the thunk is going to dispatch", but I'm not entirely sure what that means. Maybe as we begin using it the lack of typing will present itself and we can take a more informed approach.